pkg/controller/pxc/upgrade.go

```diff
@@ -364,6 +368,37 @@ func getPodOrderInSts(stsName string, podName string) (int, error) {
 	return strconv.Atoi(podName[len(stsName)+1:])
 }
 
+func (r *ReconcilePerconaXtraDBCluster) waitHostgroups(cr *api.PerconaXtraDBCluster, sfsName string, pod *corev1.Pod, waitLimit int, logger logr.Logger) error {
+	if cr.Spec.HAProxy != nil && cr.Spec.HAProxy.Enabled {
+		time.Sleep(5 * time.Second)
+		return nil
+	}
+
+	database, err := r.proxyDB(cr)
+	if err != nil {
+		return errors.Wrap(err, "failed to get proxySQL db")
+	}
+
+	defer database.Close()
+
+	podNamePrefix := fmt.Sprintf("%s.%s.%s", pod.Name, sfsName, cr.Namespace)
+
+	return retry(time.Second*10, time.Duration(waitLimit)*time.Second,
+		func() (bool, error) {
+			hostgroups := []queries.HostgroupName{queries.WriterHostgroup, queries.ReaderHostgroup}
+			present, err := database.PresentInHostgroups(hostgroups, podNamePrefix, pod.Name+"."+cr.Name+"-pxc."+cr.Namespace)
```

Comment:
1. why it is needed to define `hostgroups` names in `queries.go`, after that hardcode `hostgroups` variable on in `upgrade.go` and send constant `hostgroups` names to `queries.go` from `upgrade.go`? why not define hostgroups in `upgrade.go`?
2. why we send the same Pod Name two times into `PresentInHostgroups` function?

Reply:
some similar function https://github.com/percona/percona-xtradb-cluster-operator/blob/f2cc337993e174ccebf40d00174a5a200cd6bcba/pkg/controller/pxc/upgrade.go#L419